lychee-lib/src/utils/request.rs

```diff
@@ -278,10 +276,155 @@ mod tests {
     #[test]
     fn test_fragment_url_resolution() {
         let base = Some(Base::try_from("https://example.com/path/page.html").unwrap());
-        let input = create_input(r##"<a href="#fragment">Fragment Link</a>"##, FileType::Html);
+        let source = InputSource::String(String::new());
+
+        let uris = vec![RawUri::from("#fragment")];
+        let requests = create(uris, &source, &None, &base, &None);
+
+        assert_eq!(requests.len(), 1);
+        assert!(requests
+            .iter()
+            .any(|r| r.uri.url.as_str() == "https://example.com/path/page.html#fragment"));
+    }
+
+    #[test]
+    fn test_relative_url_resolution_from_root_path() {
```

Comment:
copied and modified the 5 tests above
* one set of copies named `*_from_root_path`
* one set of copies named `*_from_root_path_and_base_url`